include/SDL3/SDL_process.h

```diff
@@ -0,0 +1,309 @@
+#define SDL_PROP_PROCESS_STDIN_POINTER  "SDL.process.stdin"
+#define SDL_PROP_PROCESS_STDOUT_POINTER "SDL.process.stdout"
+#define SDL_PROP_PROCESS_STDERR_POINTER "SDL.process.stderr"
+
+/**
+ * Read all the output from a process.
+ *
+ * If a process was created with I/O enabled, you can use this function to read the output. This function blocks until the process is complete, capturing all output, and providing the process exit code.
+ *
+ * This is just a convenience function. If you need more control over the process, you can get the output stream from the process properties and read it directly.
+ *
+ * The data is allocated with a zero byte at the end (null terminated) for
+ * convenience. This extra byte is not included in the value reported via
+ * `datasize`.
+ *
+ * The data should be freed with SDL_free().
+ *
+ * \param process The process to read.
+ * \param datasize a pointer filled in with the number of bytes read, may be NULL.
+ * \param exitcode a pointer filled in with the process exit code if the process has exited, may be NULL.
+ * \returns the data or NULL on failure; call SDL_GetError() for more
+ *          information.
+ *
+ * \sa SDL_CreateProcess
+ * \sa SDL_CreateProcessWithProperties
+ * \sa SDL_GetProcessProperties
+ * \sa SDL_WriteProcess
+ * \sa SDL_DestroyProcess
+ *
+ * \threadsafety This function is not thread safe.
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern SDL_DECLSPEC void * SDLCALL SDL_ReadProcess(SDL_Process *process, size_t *datasize, int *exitcode);
+
+/**
+ * Write to a process.
+ *
+ * If a process was created with I/O enabled, you can use this function to send data as input to the process. This function blocks until the data is written.
+ *
+ * This is just a convenience function. If the process is structured so it takes large amounts of input and generates lots of output, you should get the input and output streams from the process properties and handle them simultaneously to prevent the process from being blocked waiting for I/O.
+ *
+ * \param process The process to write.
+ * \param ptr a pointer to a buffer containing data to write.
+ * \param size the number of bytes to write.
+ * \param closeio if SDL_TRUE, closes the process input before returning,
+ *                even in the case of an error.
+ * \returns SDL_TRUE on success or SDL_FALSE on failure; call SDL_GetError()
+ *          for more information.
+ *
+ * \sa SDL_CreateProcess
+ * \sa SDL_CreateProcessWithProperties
+ * \sa SDL_GetProcessProperties
+ * \sa SDL_ReadProcess
+ * \sa SDL_DestroyProcess
+ *
+ * \threadsafety This function is not thread safe.
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern SDL_DECLSPEC SDL_bool SDLCALL SDL_WriteProcess(SDL_Process *process, const void *ptr, size_t size, SDL_bool closeio);
+
+/**
+ * Stop a process.
+ *
+ * \param process The process to stop.
+ * \param force Stop the process immediately, without giving it a chance to free its resources.
```

Comment:
Would it be good to add some wording like "Whether to stop the process.." or "true to stop the process ...; false otherwise"? I stopped for a second when reading it because I'm used to see the opposite option explicitly mentioned in some way.

Reply:
Sure, go ahead and suggest better wording?

Reply:
`\param force If true, stop the process immediately; if false, try to stop the process gracefully. This should be true only in cases where the process is unresponsive, as force-killing will leave the child process' resources open.`

Does that sound okay?